### ISSUE TITLE
dia.ElementView:  port node with `ref` does not update correctly on size change

### ISSUE THREAD

**Issue by kumilingus** [clientIO/joint#2026]
### Discussed in https://github.com/clientIO/joint/discussions/2025

<div type='discussions-op-text'>

<sup>Originally posted by **djimenez-developair** February  7, 2023</sup>
Hello, I am trying to put a background rectangle to the port labels text, I use [this](https://codepen.io/djimenez-developair/pen/KKBYeQV) demo Wednesday as reference to get it. The problem is that I want to do this for elements that scale (concretly using FreeTransform plugin). I have added an event at the end of the codepen (line 462) so that when an element is clicked it scales it (the amount apparently does not matter) and as you can see, after scaling, the background of the label does not adjust correctly.

Any idea how I can do this? Ideally using FreeTransform plugin.

Thank you in advance!</div>

**Comment by kumilingus** [comment]
This points to several problems:
1. Ports are being **rendered twice** on resize, the second run yields a different result due to the transformations applied to port nodes.
2. The **port label layout** sets the `text-anchor` attribute (which affects the bounding box of the `<text/>`) too late. After the `ref` bounding box is measured.
3. The **port label layout** sets `y` attribute to affect the vertical text position on the wrapping group of the port label `<g/>` which has no impact. *(Note: the wrapping `<g/>` is added implicitly when the port label is made of 2 or more tags)*.

---

 As workaround, we can set `useCSSSelectors` to `false` (which solves problem **1**)
*Note: this will become a default at some point*

```js
joint.config.useCSSSelectors = false
```

This will also result in:
```js
element.attr([selector, 'fill'], 'red');
```
The `selector` must be a JointJS selector, and not CSS selector
`'label'` 👍 
`'portBody'` 👍  
`'text:first'` ⛔  
`'.my-label'` ⛔ 

And we have to set `text-anchor` and `y` attribute directly on the `<text/>` ourselves (which solves **2** & **3**).

Here's an [example](https://jsfiddle.net/kumilingus/L8x4c1p5/).

**Comment by kumilingus** [comment]
Fixed by #2031 , #2034 and #2040.
